src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,65 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendarGap;
+  max-width: $calendarSize;
+  padding: $calendarPadding;
+  margin-left: auto;
+  margin-right: auto;
+
+  $days:
+    "mon" 1,
+    "tue" 2,
+    "wed" 3,
+    "thu" 4,
+    "fri" 5,
+    "sat" 6,
+    "sun" 7;
+
+  @each $name, $dayNr in $days {
+    &--start-day-#{$name} &__day:first-child {
+      margin-left: ($daySize + $calendarGap) * ($dayNr - 1);
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} &__day:nth-last-of-type(-n + #{31 - $i}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    width: $daySize;
+    height: $daySize;
+    border: 1px solid black;
+    background-color: $calendarBg;
+    position: relative;
+    transition: all 0.5s ease-in;
```

Comment:
List all properties that you apply transition to. It will be easier to understand which properties are affected by it and also there will be no case in which something that should not have a transition has one